linux/LinuxProcessTable.c

```diff
@@ -1485,9 +1485,9 @@ static bool LinuxProcessTable_recurseProcTree(LinuxProcessTable* this, openat_ar
       {
          char* endptr;
          unsigned long parsedPid = strtoul(name, &endptr, 10);
-         if (parsedPid == 0 || parsedPid == ULONG_MAX || *endptr != '\0')
+         if (parsedPid == 0 || parsedPid >= INT_MAX || *endptr != '\0')
```

Comment:
The logic doesn't look right here. Why would `INT_MAX` become an invalid PID anyway? If we want to check for overflow, it could be `parsedPid > INT_MAX || !errno`.

Reply:
`strtoul` returns `unsigned long`, but the result is later assigned to a field of `unsigned int`. Thus anything greater than `UINT_MAX` causes an integer overflow. Everything equal to `UINT_MAX` is the sentinel value for "invalid user". Thus `parsedPid >= UINT_MAX` to catch both cases at once.

Reply:
@BenBE This is PID, not UID. Please get the right context. POSIX didn't have a clear limit on the maximum value of PID, or that PID == INT_MAX is always invalid. So I want to play safe here.

Looking at [this Stack Exchange question](https://unix.stackexchange.com/questions/16883/), it look like Linux currently has a hard limit of 4194303 (the maximum PID is *one less* than the value in `/proc/sys/kernel/pid_max`). And that no other OS had been using INT_MAX as maximum PID yet, so treating INT_MAX as invalid value may be fine, for now.

Reply:
By the way, in Unix, `pid_t` is always a signed type, unlike `uid_t`. The `pid_t` is required to be signed as [fork(2)](https://man7.org/linux/man-pages/man2/fork.2.html) returns value in this type and that negative values are reserved for special uses.